components/calendly_v2/calendly_v2.app.mjs

```diff
@@ -1,11 +1,213 @@
+      optional: true,
+    },
+    maxResults: {
+      type: "integer",
+      label: "Max Results",
+      description: "The number of rows to return",
+      optional: true,
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _baseUri() {
+      return "https://api.calendly.com";
+    },
+    _buildUserUri(user) {
+      return `${this._baseUri()}/users/${user}`;
+    },
+    _buildEventType(eventType) {
+      return `${this._baseUri()}/event_types/${eventType}`;
+    },
+    _getDefaultResponse() {
+      return {
+        collection: [],
+        pagination: {
+          count: 0,
+        },
+      };
+    },
+    _extractNextPageToken(nextPage) {
+      return nextPage
+        ? nextPage.split("page_token=")[1].split("&")[0]
+        : null;
+    },
+    _makeRequestOpts(opts) {
+      const path = opts.path ?? "";
+      const method = opts.method ?? "get";
+      const params = opts.params ?? {};
+      const headers = {
+        "Content-Type": "application/json",
+        "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
+      };
+      return {
+        url: this._baseUri() + path,
+        method,
+        headers,
+        params,
+      };
+    },
+    async _makeAsyncOptionsRequest({
+      prevContext,
+      requestType,
+    }) {
+      const defaultParams = {
+        count: 20,
+      };
+      const { nextPageParameters = defaultParams } = prevContext;
+      let response = await this[requestType](nextPageParameters);
+      return {
+        options: response.collection.map((e) => ({
+          label: e.name,
+          value: e.uri.split("/").pop(),
+        })),
+        context: {
+          nextPageParameters: response.pagination.next_page,
+        },
+      };
+    },
+    async _makeRequest(opts) {
+      const response = this._getDefaultResponse();
+      const {
+        paginate = false,
+        maxResults = 1000,
+      } = opts.params;
+      delete opts.params.paginate;
+      delete opts.params.maxResults;
+
+      do {
+        let res = await axios(
+          this,
+          this._makeRequestOpts(opts),
+        );
+        response.collection.push(...res.collection);
+        response.pagination.count += res.pagination.count;
+        response.pagination.next_page = res.pagination.next_page;
+        opts.params.page_token = this._extractNextPageToken(res.pagination.next_page);
+      } while (paginate && opts.params.page_token && response.collection.length < maxResults);
+
+      if (response.collection.length > maxResults) {
+        response.collection.length = maxResults;
+        response.pagination.count = maxResults;
+      }
+
+      return response;
+    },
+    async getUserInfo(user = "me") {
+      const opts = {
+        path: `/users/${user}`,
+      };
+      return await axios(
+        this,
+        this._makeRequestOpts(opts),
+      );
+    },
```

Comment:
Pass in `$` to be used in the axios request.
Also, you can remove the `await` since it's the last line of an `async` function.
```suggestion
    async getUserInfo($, user = "me") {
      const opts = {
        path: `/users/${user}`,
      };
      return axios(
        $ ?? this,
        this._makeRequestOpts(opts),
      );
    },
```

Reply:
Issue addressed